mps_youtube/players/other_players.py

```diff
@@ -0,0 +1,254 @@
+import abc
+import abc
+import os
+import sys
+import random
+import tempfile
+import subprocess
+import logging
+import json
+import re
+import socket
+import math
+import time
+import shlex
+from urllib.error import HTTPError, URLError
+
+from .. import g, screen, c, streams, history, content, paths, config, util
+from ..player import Player
+
+class OtherPlayers(Player):
+
+    def play_range(self, songlist, shuffle=False, repeat=False, override=False):
+        """ Play a range of songs, exit cleanly on keyboard interrupt. """
+        if shuffle:
+            random.shuffle(songlist)
+
+        n = 0
+        while 0 <= n <= len(songlist)-1:
+            song = songlist[n]
+            g.content = self._playback_progress(n, songlist, repeat=repeat)
+
+            if not g.command_line:
+                screen.update(fill_blank=False)
+
+            hasnext = len(songlist) > n + 1
+
+            if hasnext:
+                streams.preload(songlist[n + 1], override=override)
+
+            if config.SET_TITLE.get:
+                util.set_window_title(song.title + " - mpsyt")
+            try:
+                returncode = self._playsong(song, override=override)
+
+            except KeyboardInterrupt:
+                logging.info("Keyboard Interrupt")
+                util.xprint(c.w + "Stopping...                          ")
+                screen.reset_terminal()
+                g.message = c.y + "Playback halted" + c.w
+                raise KeyboardInterrupt
+                break
+            if config.SET_TITLE.get:
+                util.set_window_title("mpsyt")
+
+            if returncode == 42:
```

Comment:
These two magic return codes only apply to mpv/mplayer (using the special configuration mpsyt provides).